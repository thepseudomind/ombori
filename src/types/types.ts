export interface IMain{
    status: boolean,
    toggleSidebar?: any
}

export interface IProduct{
    id: number,
    email: string,
    first_name: string,
    last_name: string,
    avatar: string
}

export type TLoader = {
    mini: boolean
}

export type TMessage = {
    message: string
}