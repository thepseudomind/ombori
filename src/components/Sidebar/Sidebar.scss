@import '../../mixins/mixins';

.sidebar{
    background-color: white;
    position: fixed;
    top: 0;
    right: 0;
    height: 100vh;
    width: 0%;
    box-shadow: 0 .1rem 3rem rgba(black, .1);
    text-transform: uppercase;
    transition: all .5s ease;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    z-index: 3;

    & > *{
        width: 80%;
        margin: 0 auto;
    }

    &.active{
        width: 25%;

        @include respond(tab-land){
            width: 35%;
        }
    
        @include respond(phone){
            width: 45%;
        }
    }

    &.active &__close{
        opacity: 1;
        visibility: visible;
        transition: all .5s ease;
    }

    &__close{
        font-size: 6rem;
        font-weight: 100;
        font-family: 'Montserrat', sans-serif;
        cursor: pointer;
        opacity: 0;
        visibility: hidden;
    }

    &__title{
        font-size: 2rem;
    }

    &__heading{
        display: flex;
        justify-content: space-between;
        align-items: center;
    }

    &__content{
        font-size: 2rem;
    }

    &__list{
        &--item{
            list-style: none;
            font-family: 'Montserrat', sans-serif;
            font-size: 1.5rem;
            text-transform: uppercase;
            cursor: pointer;

            &:not(:last-child){
                margin: 1rem 0;
            }
        }
    }
}