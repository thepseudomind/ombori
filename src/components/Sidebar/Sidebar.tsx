import React from 'react';
import './Sidebar.scss';
import { IMain } from '../../types/types';

const Sidebar = ({status, toggleSidebar} : IMain)=>{
    return (
        <div className={`sidebar${(status) ? ' active' : ''}`}>
            <div className="sidebar__heading">
                <h2 className="sidebar__title">welcome, marija</h2>
                <a className="sidebar__close" onClick={()=> toggleSidebar(false)}>&times;</a>
            </div>
            <div className="sidebar__content">
                <h2 className="sidebar__title">menu</h2>
                <ul className="sidebar__list">
                    <li className="sidebar__list--item">home</li>
                    <li className="sidebar__list--item">friends</li>
                    <li className="sidebar__list--item">profile</li>
                    <li className="sidebar__list--item">settings</li>
                    <li className="sidebar__list--item">logout</li>
                </ul>
            </div>
            <div className="sidebar__heading">
                <h2 className="sidebar__title">built with&nbsp;<span>♥️</span>&nbsp;&nbsp;by ThePseudoMind</h2>
            </div>
        </div>
    );
}

export default Sidebar;