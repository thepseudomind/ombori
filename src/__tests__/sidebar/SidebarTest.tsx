import React from 'react';
import { shallow } from  'enzyme';
import Sidebar from '../../components/Sidebar/Sidebar';
import { IMain } from '../../types/types';

it('there is button to click', ()=>{
    const mockProps : IMain = {
        status: true,
        toggleSidebar: ()=>{}
    };
    const wrapper = shallow(<Sidebar {...mockProps}/>);
    wrapper.find('.sidebar__close').simulate('click');
    expect(wrapper).toMatchSnapshot();
});

it('check is sidebar is active', ()=>{
    const mockProps : IMain = {
        status: true,
        toggleSidebar: ()=>{}
    };
    const wrapper = shallow(<Sidebar {...mockProps}/>);
    expect(wrapper.find('.sidebar').html()).toMatchSnapshot();
});

it('check is sidebar is inactive', ()=>{
    const mockProps : IMain = {
        status: false,
        toggleSidebar: ()=>{}
    };
    const wrapper = shallow(<Sidebar {...mockProps}/>);    
    expect(wrapper.find('.sidebar').html()).toMatchSnapshot();
});

