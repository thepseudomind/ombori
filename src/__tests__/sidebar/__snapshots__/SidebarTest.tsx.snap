// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check is sidebar is active 1`] = `"<div class=\\"sidebar active\\"><div class=\\"sidebar__heading\\"><h2 class=\\"sidebar__title\\">welcome, marija</h2><a class=\\"sidebar__close\\">×</a></div><div class=\\"sidebar__content\\"><h2 class=\\"sidebar__title\\">sort by</h2><ul class=\\"sidebar__list\\"><li class=\\"sidebar__list--item\\">none</li><li class=\\"sidebar__list--item\\">by asc</li><li class=\\"sidebar__list--item\\">by id</li><li class=\\"sidebar__list--item\\">by size</li><li class=\\"sidebar__list--item\\">by price</li></ul></div><div class=\\"sidebar__heading\\"><h2 class=\\"sidebar__title\\">built with <span>♥️</span>  by ThePseudoMind</h2></div></div>"`;

exports[`check is sidebar is inactive 1`] = `"<div class=\\"sidebar\\"><div class=\\"sidebar__heading\\"><h2 class=\\"sidebar__title\\">welcome, marija</h2><a class=\\"sidebar__close\\">×</a></div><div class=\\"sidebar__content\\"><h2 class=\\"sidebar__title\\">sort by</h2><ul class=\\"sidebar__list\\"><li class=\\"sidebar__list--item\\">none</li><li class=\\"sidebar__list--item\\">by asc</li><li class=\\"sidebar__list--item\\">by id</li><li class=\\"sidebar__list--item\\">by size</li><li class=\\"sidebar__list--item\\">by price</li></ul></div><div class=\\"sidebar__heading\\"><h2 class=\\"sidebar__title\\">built with <span>♥️</span>  by ThePseudoMind</h2></div></div>"`;

exports[`there is button to click 1`] = `
<div
  className="sidebar active"
>
  <div
    className="sidebar__heading"
  >
    <h2
      className="sidebar__title"
    >
      welcome, marija
    </h2>
    <a
      className="sidebar__close"
      onClick={[Function]}
    >
      ×
    </a>
  </div>
  <div
    className="sidebar__content"
  >
    <h2
      className="sidebar__title"
    >
      sort by
    </h2>
    <ul
      className="sidebar__list"
    >
      <li
        className="sidebar__list--item"
      >
        none
      </li>
      <li
        className="sidebar__list--item"
      >
        by asc
      </li>
      <li
        className="sidebar__list--item"
      >
        by id
      </li>
      <li
        className="sidebar__list--item"
      >
        by size
      </li>
      <li
        className="sidebar__list--item"
      >
        by price
      </li>
    </ul>
  </div>
  <div
    className="sidebar__heading"
  >
    <h2
      className="sidebar__title"
    >
      built with 
      <span>
        ♥️
      </span>
        by ThePseudoMind
    </h2>
  </div>
</div>
`;
